--- a/Makefile	2024-01-30 14:17:52.033584575 +0000
+++ b/Makefile	2024-01-30 14:19:34.070251184 +0000
@@ -1,4 +1,5 @@
 # Try to locate the AWS SDK if not specified with AWS_SDK_PATH
+MACHINE := $(shell gcc -dumpmachine)
 ifeq ($(AWS_SDK_PATH),)
   ifneq ($(wildcard /usr/include/aws),)
     AWS_SDK_PATH := /usr
@@ -15,6 +16,8 @@
     AWS_SDK_LIB_PATH := $(addsuffix /lib,$(AWS_SDK_PATH))
   else ifneq ($(wildcard $(AWS_SDK_PATH)/lib64/libaws-c-common.*),)
     AWS_SDK_LIB_PATH := $(addsuffix /lib64,$(AWS_SDK_PATH))
+  else ifneq ($(wildcard $(AWS_SDK_PATH)/lib/$(MACHINE)/libaws-c-common.*),)
+    AWS_SDK_LIB_PATH := $(addsuffix /lib/$(MACHINE),$(AWS_SDK_PATH))
   else
     $(error neither lib or lib64 found in AWS SDK)
   endif
